site/documentation/references/security/index.md

```diff
@@ -0,0 +1,288 @@
+must be provided on startup. The authorization policy must be structured as follows:
+
+```ruby
+# The "system" namespace is reserved for internal use
+# by OPA. Authorization policy must be defined under
+# system.authz as follows:
+package system.authz
+
+default allow = false  # Reject requests by default.
+
+allow {
+  # Logic to authorize request goes here.
+}
+```
+
+When OPA receives a request, it executes a query against the document defined
+`data.system.authz.allow`. The implementation of the policy may span multiple
+packages however it is recommended that administrators keep the policy under the
+`system` namespace.
+
+If the document produced by the ``allow`` rule is ``true``, the request is
+processed normally. If the document is undefined or **not** ``true``, the
+request is rejected immediately.
+
+OPA provides the following input document when executing the authorization
+policy:
+
+```ruby
+{
+    "input": {
+        # Identity established by authentication scheme.
+        # When Bearer tokens are used, the identity is
+        # set to the Bearer token value.
+        "identity": <String>,
+
+        # One of {GET, POST, PUT, PATCH, DELETE}.
+        "method": <HTTP Method>,
+
+        # URL path represented as an array.
+        # For example: /v1/data/exempli-gratia
+        # is represented as [v1, data, exampli-gratia]
+        "path": <HTTP URL Path>,
+    }
+}
+```
+{: .opa-collapse--ignore}
+
+At a minimum, the authorization policy should grant access to a special root
+identity:
+
+```ruby
+package system.authz
+
+default allow = false          # Reject requests by default.
+
+allow {                        # Allow request if...
+    input.identity = "secret"  # Identity is the secret root key.
+}
+```
+{: .opa-collapse--ignore}
+
+When OPA is configured with this minimal authorization policy, requests without
+authentication are rejected:
+
+```http
+GET /v1/policies HTTP/1.1
+```
+
+Response:
+
+```http
+HTTP/1.1 401 Unauthorized
+Content-Type: application/json
+```
+
+```json
+{
+  "code": "unauthorized",
+  "message": "request rejected by administrative policy"
+}
+```
+
+However, if Bearer token authentication is enabled and the request includes the
+secret from above, the request is allowed:
+
+```http
+GET /v1/policies HTTP/1.1
+Authorization: Bearer secret
+```
+
+Response:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+```
+
+When Bearer tokens are used for authentication, the policy should at minimum
+validate the identity:
+
+```ruby
+package system.authz
+
+default allow = false          # Reject requests by default.
+
+allow {                        # Allow request if...
+    input.identity = "secret"  # Identity is the secret root key.
+}
+
+allow {                        # Allow request if...
+    tokens[input.identity]     # Identity exists in "tokens".
+}
+
+# Tokens may defined in policy or pushed into OPA as data.
```

Comment:
At first, I scrolled back up to see if `tokens` had been defined before. Maybe place this directly after the import statements?